mdp/nodes/recursive_expansion_nodes.py

```diff
@@ -0,0 +1,767 @@
+    of the first kind.
+    The recursion formula is
+    T_n = 2xT_{n-1} - T_{n-2}.
+
+    .. note::
+
+        The procedure on how an recursion function is build can be found in the
+        docstring of the recursion function for
+        standard polynomials *recf_standard_poly*.
+
+
+    :param result: Contains the observations along the first dimension
+        and the different function values of expansion w.r.t. an observation
+        along the second dimension.
+    :type result: numpy.ndarray
+
+    :param x: The data to be expanded.
+    :type x: numpy.ndarray
+
+    :param special: Index of a special element to be considered in the
+        recursion formula. E.g. the first order function value.
+    :type special: int
+
+    :param n: The order of the function to be computed at this step.
+    :type n: int
+
+    :param cur_var: The index of the current variable to be considered in the
+        recursion. This value will have to be lower than x.shape[1].
+    :type cur_var: int
+
+    :param pos: The index of the element to be computed, along the second
+        dimension of result.
+    :type pos: int
+
+    :returns: The vectorized result (along the observations) of the
+        n-th recursion step of the cur_var-th variable.
+    :rtype: numpy.ndarray
+
+    .. admonition:: Reference
+
+        https://en.wikipedia.org/wiki/Chebyshev_polynomials
+    """
+    return 2. * x[:, cur_var] * result[:, pos-1] - result[:, pos-2]
+
+
+# collect the recusions and set the corresponding domains
+recfs = {'standard_poly': (init_standard_poly, recf_standard_poly,
+                           -float('Inf'), float('Inf')),
+         'legendre_poly': (init_legendre_poly, recf_legendre_poly,
+                           -1, 1),
+         'legendre_rational': (init_legendre_rational,
+                               recf_legendre_rational, 0, float('Inf')),
+         'chebyshev_poly': (init_chebyshev_poly,
+                            recf_chebyshev_poly, -1, 1)}
+
+
+def process(off, leadVar, leadDeg, deg, pos, result, todoList):
+    """Computes an inplace not exactly tensorproduct of the given
+    basis functions (only the simplex of the single variable bases,
+    as we limit the multi variable polynomials by a given maximum degree).
+    """
+    for v in range(leadVar):
+        l = deg-leadDeg
+        k = 1+deg*v
+        for d in range(1, deg-leadDeg+1):
+            l = deg+1-leadDeg-d
+            result[:, pos:pos+l] = result[:, 1+off:1+off+l] * result[:, k:k+1]
+            if l > 1 and leadVar > 0:
+                todoList.append((pos-1, v, leadDeg+d))
+
+            pos += l
+            k += 1
+            l -= 1
+
+    return pos
+
+
+class RecursiveExpansionNode(PolynomialExpansionNode):
+    """Recursively computable (orthogonal) expansions.
+
+    .. attribute:: lower
+
+        The lower bound of the domain, on which the recursion
+        function sequence selected, is defined or orthogonal.
```

Comment:
The commas here are wrong, because "on which the recursion function sequence selected" is defining and the other clause too. However I'd simplify the whole thing by just remove 'sequence selected':

The lower bound of the domain on which the recursion function is defined or orthogonal.